rllib/utils/actor_manager.py

```diff
@@ -0,0 +1,324 @@
+from collections import defaultdict
+import logging
+from typing import Callable, List, Optional, Tuple, TypeVar
+
+import ray
+from ray.actor import ActorHandle
+from ray.exceptions import RayActorError
+from ray.util.annotations import DeveloperAPI
+
+
+logger = logging.getLogger(__name__)
+# Generic type var for foreach_* methods.
+T = TypeVar("T")
+
+
+class FaultTolerantActorManager:
+    """A manager that is aware of the healthiness states of remote actors.
+
+    Args:
+        actors: A list of ray remote actors to manage on. These actors must have an
+            ``apply`` method which takes a function with no parameters.
```

Comment:
Wait: Takes a `Callable[[Any], None]`, correct?

Meaning the lambda must have access to the actual object(!) behind the ActorHandle. Otherwise, it won't be able to do anything :)